databases/foundationdb/Portfile

```diff
@@ -0,0 +1,92 @@
+platforms           darwin
+description         FoundationDB is a distributed database providing ACID transactions.
+long_description    \
+    FoundationDB is a scalable, fault-tolerant, ordered \
+    key-value store with full ACID transactions.  This port \
+    will install version ${version} of the FoundationDB \
+    binaries suitable for development on macOS.
+
+homepage            https://www.foundationdb.org/
+
+github.setup        apple foundationdb ${version}
+
+checksums           rmd160  2bf9a258d5db140d232fb6355fc810010824c014 \
+                    sha256  034f69d4ddf7faccd7cae2a6438d64643ac98a3a9c7b3e6ed2a03105c480df95 \
+                    size    9694306
+
+depends_lib-append  port:mono
+
+cmake.generator     Ninja
+build.jobs          1
+
+destroot {
+    # install compiled binaries
+
+    xinstall -d -m 755 ${destroot}${prefix}/bin
+    xinstall -m 755 ${workpath}/build/packages/bin/fdbcli \
+        ${workpath}/build/packages/bin/fdbbackup \
+        ${destroot}${prefix}/bin
+    foreach x {backup_agent dr_agent fastrestore_tool fdbdr fdbrestore} {
+        ln -f -s fdbbackup ${destroot}${prefix}/bin/${x}
+    }
+    
+    xinstall -d -m 755 ${destroot}${prefix}/lib
+    xinstall -m 755 ${workpath}/build/packages/lib/libfdb_c.dylib \
+        ${destroot}${prefix}/lib
+
+    xinstall -d -m 755 ${destroot}${prefix}/libexec
+    xinstall -m 755 ${workpath}/build/packages/bin/fdbmonitor \
+        ${destroot}${prefix}/libexec
+
+    xinstall -d -m 755 ${destroot}${prefix}/sbin
+    xinstall -m 755 ${workpath}/build/packages/bin/fdbserver \
+        ${destroot}${prefix}/sbin
+
+    # install header files
+
+    xinstall -d -m 755 ${destroot}${prefix}/include/foundationdb
+    xinstall -m 755 ${worksrcpath}/bindings/c/foundationdb/fdb_c.h \
+        ${workpath}/build/bindings/c/foundationdb/fdb_c_options.g.h \
+        ${worksrcpath}/fdbclient/vexillographer/fdb.options \
+        ${destroot}${prefix}/include/foundationdb
+
+    # install sample config files and adjust paths
+
+    xinstall -d -m 755 ${destroot}${prefix}/etc/foundationdb
+    xinstall -m 755 ${worksrcpath}/packaging/osx/foundationdb.conf.new \
+        ${destroot}${prefix}/etc/foundationdb
+
+    reinplace "s|/usr/local/etc/|${prefix}/etc/|g" ${destroot}${prefix}/etc/foundationdb/foundationdb.conf.new
+    reinplace "s|/usr/local/libexec/|${prefix}/sbin/|g" ${destroot}${prefix}/etc/foundationdb/foundationdb.conf.new
+    reinplace "s|/usr/local/foundationdb/backup_agent/backup_agent|${prefix}/bin/backup_agent|g" ${destroot}${prefix}/etc/foundationdb/foundationdb.conf.new
+    reinplace "s|/usr/local/foundationdb/|${prefix}/foundationdb/|g" ${destroot}${prefix}/etc/foundationdb/foundationdb.conf.new
+}
+
+post-activate {
+    
+    # link foundationdb client files to /usr/local
+    file mkdir /usr/local/lib
+    ln -f -s ${prefix}/lib/libfdb_c.dylib /usr/local/lib/
+    file mkdir /usr/local/include
+    ln -f -s ${prefix}/include/foundationdb /usr/local/include/
```

Comment:
Writing outside the install prefix is not really allowed (apart from a few special case which this not one).

in particular we don’t want any interactions with /usr/local, so you should remove this. Why was it needed ?

Reply:
The symlinks into /usr/local were a misguided effort to allow the Python API module to locate the FoundationDB library. Today I found that the preferred method is to place a symbolic link to the library in the same directory as the Python module: https://github.com/apple/foundationdb/blob/d3600696ce26ee860d48fed7dd1d3cfb56688d1e/bindings/python/fdb/impl.py#L1294

Reply:
thanks.